### ISSUE TITLE
Expose file descriptor flag

### ISSUE THREAD

**Issue by marcbachmann** [flumedb/flumelog-offset#3]
I want to set the file descriptor flag to `r` to prevent any writes.
There are two use cases I'm interested in. I'd like to have an archiver, which doesn't have access in the separate process. The other use case would be to prevent writes in another process which creates projections (edit: oh, to create a projection in a separate process we would also need to expose the value `blocks.offset`).
The cache is also not needed for those use cases.

Can we expose that config or should I just wrap the module to prevent developers from doing writes? Explicitly setting it would be better IMO.

**Comment by marcbachmann** [comment]
e.g.
```js
pull(
  log.stream({live: true}),
  pull.drain(function (e) {console.log(e.seq)}, console.log)
)

const update = _.debounce(updateOffset, 100)
fs.watch(projects.file, (eventType) => {
  if (eventType === 'change') update()
})

function updateOffset () {
  // that restore doesn't work because of the additional block that is saved.
  // we would need to apply frame.restore from https://github.com/flumedb/flumelog-offset/blob/master/frame/recoverable.js#L46
  fs.stat(projects.file, function (err, stat) {
    if (err) throw err
    log.since.set(Math.max(log.since.value, stat.size - 1024*4))
  })
}
```

**Comment by dominictarr** [comment]
haha, yes I was just about to describe that code you posted. Yes I think this would be a execellent feature. hmm, I think the codec, block size, etc should all be moved to a options {} object.

**Comment by marcbachmann** [comment]
😁 Yep, I think so too. I could do a pr somewhen this week. Have to go to sleep now.

**Comment by dominictarr** [comment]
looking forward to it!

**Comment by marcbachmann** [comment]
Merged in #4
